### PR TITLE
Set up Enzyme tests

### PR DESCRIPTION
**Pull request by k3ntako**
Chose Enzyme/Mocha/Chai over Jest because Enzyme allows us to use the libraries we are already using (i.e., Chai, Mocha, Sinon). Keeping the libraries consistent helps us transition from writing front-end tests to back-end tests. Mastering one will help with the other, while learning to use Jest would suck up time without helping us with writing better back-end tests.

Disadvantages:
1. Jest is a test runner, assertion library, and mocking library. 
  - Jest may have been easier to set up.
  - Any updates to a part of Jest will consider the other parts. While Chai's top priority is not working with Enzyme.

Challenges/Things to consider: 
1. Components are nested in Providers. 
  - Mostly solved (just not pretty).
2. Naming components.
  - For example: I cannot use Enzyme's `.find()` to find `Table` on the roster page. This is because we wrap React-Toolbox's Table in our own `Table` component. We should consider naming our component something else.
  - However, it will likely be difficult to always avoid naming collisions.
3. Need to figure out how to deal with state.
  - ~~Working to figure this out in this PR.~~

Updates:
1. CircleCI: Updated Node version to `lts`.
  - `jsdom` requires Node `8.0` or greater, and we had `7.10`.
2. TestHelper: 
  - `createPerson()` is used to create an instance of a `Person` and to populate a team.
    - Does not rely on random number, so it's a pure function.
  - `createId()` will make sure that all `_id`s are unique across all tests. 
    - This is the only state held in TestHelper. Everything else should be held in their respective test files.
  - `createTeam()` - creates a team with the same ID every time.
3. File organization: I thought we could mirror the file structure of our Components/Pages. 
4. Started by writing test for Roster component.

### CODE REVIEW

**Comment by mienaikoe** [comment]
> While Chai's top priority is not working with Enzyme.
What do you mean by this?

> We should consider naming our component something else.
Yea I think most of our components are wrappers for specific reasons. I think an exception to this would be <Button/>, which is wrapped for styling reasons. Why can't you test for the existence of either component? If one exists, then the other ought to. If it's searching by string, can we use `displayName`?

**Comment by k3ntako** [comment]
> > While Chai's top priority is not working with Enzyme.
> > What do you mean by this?
> 
> > We should consider naming our component something else.
> > Yea I think most of our components are wrappers for specific reasons. I think an exception to this would be , which is wrapped for styling reasons. Why can't you test for the existence of either component? If one exists, then the other ought to. If it's searching by string, can we use `displayName`?

I just thought updates to Jest would be smoother because if an update in Jest's  mocking library breaks something in the mocking library, they're more likely to notice it before they release that update. 

What's the exception? I'm just seeing a space.
I wanted to know how many tables there were. Coaches should have 3 (or 2 if no invitees) while members should have 2.

**Comment by mienaikoe** [comment]
Ohh I see so you might have 6 instead of 3.

So apprently github doesn't sanitize html on comment inputs. Does this mean I can embed arbitrary code?

<iframe src="http://www.google.com"/>

Huh.. Well anyway, it should have said `<Button/>`

**Comment by k3ntako** [comment]
> Ohh I see so you might have 6 instead of 3.
> 
> So apprently github doesn't sanitize html on comment inputs. Does this mean I can embed arbitrary code?
> 
> <iframe src="http://www.google.com"/>
> Huh.. Well anyway, it should have said `<Button/>`

Yeah 6 instead of 3.
ohh okay. Yeah I was thinking Buttons will be tricky.

**Comment by k3ntako** [comment]
Useful links
Setting Up Enzyme Tests
https://medium.com/riipen-engineering/testing-react-with-enzyme-part-1-setup-ff49e51f8ff0
https://airbnb.io/enzyme/docs/guides/mocha.html
https://airbnb.io/enzyme/docs/guides/jsdom.html

Cheat Sheet
https://devhints.io/enzyme

Getting Around Consumers
https://gist.github.com/danmakenoise/c503854508f80137fe547943218ccc05

Testing With Redux
https://hackernoon.com/unit-testing-redux-connected-components-692fa3c4441c

Updating CircleCI Node Version
https://circleci.com/docs/2.0/circleci-images/#nodejs

File: client/tests/pages/Team/Roster.spec.js [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    // Set-up Mock Redux
+    const initialState = {};
+    const mockStore = configureStore();
+    const store = mockStore({ session: { me: meAsMember } });
+
+    // <Router> is required for using anything related to withRouter or <Link>
+    wrapper = mount(
+      <MemoryRouter>
+        <Provider store={store}>
+          <RosterWithProviders />
```
Comment (mienaikoe) [inline client/tests/pages/Team/Roster.spec.js]: Don't need it with Providers now yay!

File: client/tests/setup.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+var Adapter = require('enzyme-adapter-react-16');
+
+enzyme.configure({ adapter: new Adapter() });
+
+// Create a place for react to grab onto.
+const app = document.createElement('div');
+app.id = 'app';
+document
+  .getElementsByTagName('body')
+  .item(0)
```
[the remark this answers is not in the document]
Comment (k3ntako) [reply client/tests/setup.js]: we actually don't need this. `client/index.js` isn't even being rendered, and that's the only component that relies on `<div id="app">`

**Pull request merged**